InAppUtils/InAppUtils.m

```diff
@@ -220,12 +220,18 @@ - (void)productsRequest:(SKProductsRequest *)request
 }
 
 // SKProductsRequestDelegate network error
-- (void)request:(SKRequest *)request didFailWithError:(NSError *)error{
+- (void)request:(SKRequest *)request didFailWithError:(NSError *)error
+{
     NSString *key = RCTKeyForInstance(request);
     RCTResponseSenderBlock callback = _callbacks[key];
-    if(callback) {
+    if (callback) {
+        // Ensure error.userData can be converted to JSON without error.
+        // This will remove any NSURL from the error.userData.
+        error = RCTErrorClean(error);
```

Comment:
Ideally, but upstreaming it and waiting for merge will take pretty long and till then this method will crash.

I will try upstream that change too